.github/workflows/main.yml

```diff
@@ -29,24 +31,52 @@ jobs:
           echo "DESTDIR=$REPO_NAME-release" >> $GITHUB_ENV
           echo "GST_PLUGIN_PATH=$GITHUB_WORKSPACE/builddir/lib/plugins" >> $GITHUB_ENV
 
+      - name: Set i386 environment variables
+        if: matrix.platform == 'linux-x86-full'
+        run: |
+            echo "CC=cc -m32" >> $GITHUB_ENV
+            echo "CXX=c++ -m32" >> $GITHUB_ENV
+            echo "PKG_CONFIG_PATH=/usr/lib/i386-linux-gnu/pkgconfig/" >> $GITHUB_ENV
+
       - name: Install dependencies
         run: |
           sudo apt update
+          sudo apt install --assume-yes --no-install-recommends valgrind python3-pip ninja-build
+          sudo pip3 install meson==1.0.1
+
+      - name: Install i386 dependencies
+        if: matrix.platform == 'linux-x86-full'
+        run: |
+          sudo apt install gcc-multilib g++-multilib
+          # hack to avoid that zlib x64 gets picked up
+          sudo apt remove --assume-yes zlib1g-dev
+
+      - name: Install GStreamer dev packages
+        if: matrix.platform == 'linux-x86_64'
+        run: |
           # Hack to install libunwind-dev (gstreamer dependency)
           sudo apt remove --assume-yes libunwind-14-dev
-          sudo apt install --assume-yes --no-install-recommends meson libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev libgstreamer-plugins-bad1.0-dev gstreamer1.0-plugins-bad valgrind
+          sudo apt install libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev libgstreamer-plugins-bad1.0-dev gstreamer1.0-plugins-bad
 
       - name: Configure
-        run: meson setup builddir
+        run: |
+          # Configure using python script only in full configuration
+          if $(echo "${{ matrix.platform }}" | grep -q "full"); then
+            python3 ./configure_gst_full.py builddir
+          else
+            meson setup builddir
+          fi
 
       - name: Build
         run: meson compile --verbose -C builddir
 
       - name: Test
+        if: matrix.platform == 'linux-x86_64'
```

Comment:
Any problem testing the 32-bits one as well?

Reply:
yes the test are not passing because of nvidia 32bit library not available and the valgrind 32-bits support.